crates/lox-io/src/ndm/kvn/deserializer.rs

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2023. Helge Eichhorn and the LOX contributors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use nom::error::ErrorKind;
+
+pub trait KvnDeserializer {
+    fn deserialize<'a>(
+        lines: &mut std::iter::Peekable<impl Iterator<Item = &'a str>>,
+    ) -> Result<Self, KvnDeserializerErr<&'a str>>
```

Comment:
Errors should generally be `'static`, since they may outlive their data source or be passed between threads. This is doubly true for library error types, since we can't predict exactly how users will structure their code.

`KvnDeserializerErr<String>>` is a more flexible option.